src/Compilers/Core/Portable/DiagnosticAnalyzer/DefaultAnalyzerAssemblyLoader.Core.cs

```diff
@@ -17,95 +17,43 @@ namespace Microsoft.CodeAnalysis
-        /// compiler's version of a given assembly over the analyzer's version.</p>
-        ///
-        /// <p>If we neglect to do this, then in the case where the user ships the compiler or its
-        /// dependencies in the analyzer's bin directory, we could end up loading a separate
-        /// instance of those assemblies in the process of loading the analyzer, which will surface
-        /// as a failure to load the analyzer.</p>
-        /// </summary>
-        internal static readonly ImmutableHashSet<string> CompilerAssemblySimpleNames =
-           ImmutableHashSet.Create(
-               StringComparer.OrdinalIgnoreCase,
-               "Microsoft.CodeAnalysis",
-               "Microsoft.CodeAnalysis.CSharp",
-               "Microsoft.CodeAnalysis.VisualBasic",
-               "System.Collections",
-               "System.Collections.Concurrent",
-               "System.Collections.Immutable",
-               "System.Console",
-               "System.Diagnostics.Debug",
-               "System.Diagnostics.StackTrace",
-               "System.Diagnostics.Tracing",
-               "System.IO.Compression",
-               "System.IO.FileSystem",
-               "System.Linq",
-               "System.Linq.Expressions",
-               "System.Memory",
-               "System.Reflection.Metadata",
-               "System.Reflection.Primitives",
-               "System.Resources.ResourceManager",
-               "System.Runtime",
-               "System.Runtime.CompilerServices.Unsafe",
-               "System.Runtime.Extensions",
-               "System.Runtime.InteropServices",
-               "System.Runtime.InteropServices.RuntimeInformation",
-               "System.Runtime.Loader",
-               "System.Runtime.Numerics",
-               "System.Runtime.Serialization.Primitives",
-               "System.Security.Cryptography.Algorithms",
-               "System.Security.Cryptography.Primitives",
-               "System.Text.Encoding.CodePages",
-               "System.Text.Encoding.Extensions",
-               "System.Text.RegularExpressions",
-               "System.Threading",
-               "System.Threading.Tasks",
-               "System.Threading.Tasks.Parallel",
-               "System.Threading.Thread",
-               "System.Threading.ThreadPool",
-               "System.Xml.ReaderWriter",
-               "System.Xml.XDocument",
-               "System.Xml.XPath.XDocument");
-
-        internal virtual ImmutableHashSet<string> AssemblySimpleNamesToBeLoadedInCompilerContext => CompilerAssemblySimpleNames;
-
-        // This is the context where compiler (and some of its dependencies) are being loaded into, which might be different from AssemblyLoadContext.Default.
-        private static readonly AssemblyLoadContext s_compilerLoadContext = AssemblyLoadContext.GetLoadContext(typeof(DefaultAnalyzerAssemblyLoader).GetTypeInfo().Assembly)!;
-
+        private readonly AssemblyLoadContext _compilerLoadContext;
```

Comment:
not related to this PR, consider sealing `DefaultAnalyzerAssemblyLoader` (both of them)